pkg/secrets/keyvault/client.go

```diff
@@ -78,33 +92,32 @@ func IsKeyVaultAccessible(kvsecretclient secrets.SecretClient) bool {
 	data = map[string][]byte{
 		"test": []byte(""),
 	}
-	err = kvsecretclient.Upsert(ctx, key, data)
-	if strings.Contains(err.Error(), errhelp.Forbidden) {
+	err = client.Upsert(ctx, key, data)
+	if err != nil && strings.Contains(err.Error(), errhelp.Forbidden) {
 		return false
 	}
 
-	err = kvsecretclient.Delete(ctx, key)
-	if strings.Contains(err.Error(), errhelp.Forbidden) {
+	err = client.Delete(ctx, key)
+	if err != nil && strings.Contains(err.Error(), errhelp.Forbidden) {
 		return false
 	}
 
 	return true
 }
 
+// TODO: This isn't even used except in test, maybe just delete it?
```

Comment:
What are thoughts on me removing this in favor of just a single method `Upsert`?

Reply:
I like that.

Reply:
Done